arangod/Statistics/StatisticsWorker.cpp

```diff
@@ -0,0 +1,968 @@
+  for (auto const& val : dist._counts) {
+    builder.add(VPackValue(val));
+  }
+  builder.close();
+
+  builder.close();
+
+  return builder;
+}
+
+void StatisticsWorker::_saveSlice(VPackSlice const& slice,
+                                  std::string const& collection) const {
+  if (isStopping()) {
+    return;
+  }
+
+  TRI_vocbase_t* vocbase = DatabaseFeature::DATABASE->systemDatabase();
+  arangodb::OperationOptions opOptions;
+  opOptions.isRestore = false, opOptions.waitForSync = false;
+  opOptions.returnNew = false;
+  opOptions.silent = false;
+
+  // find and load collection given by name or identifier
+  auto transactionContext(transaction::StandaloneContext::Create(vocbase));
+  SingleCollectionTransaction trx(transactionContext, collection,
+                                  AccessMode::Type::WRITE);
+  trx.addHint(transaction::Hints::Hint::SINGLE_OPERATION);
+
+  Result res = trx.begin();
+
+  if (!res.ok()) {
+    return;
+  }
+
+  arangodb::OperationResult result =
+      trx.insert(collection, slice, opOptions);
+
+  // Will commit if no error occured.
+  // or abort if an error occured.
+  // result stays valid!
+  res = trx.finish(result.code);
+
+  /* useless until we do something
+  if (result.failed()) {
+    return;
+  }
+
+  if (!res.ok()) {
+    return;
+  }*/
+}
+
+void StatisticsWorker::createCollections() const {
+  if (!ServerState::instance()->isRunningInCluster() ||
+       ServerState::instance()->isCoordinator()) {
+         _createCollection("_statisticsRaw");
+         _createCollection("_statistics");
+         _createCollection("_statistics15");
+  } else {
+    // wait for collection creation cluster db server
+    while (true) {
+      usleep(500*1000);
+
+      TRI_vocbase_t* vocbase = DatabaseFeature::DATABASE->systemDatabase();
+
+      LogicalCollection* coll = vocbase->lookupCollection("_statisticsRaw");
+      if (coll == nullptr) continue;
+
+      coll = vocbase->lookupCollection("_statistics");
+      if (coll == nullptr) continue;
+
+      coll = vocbase->lookupCollection("_statistics15");
+      if (coll == nullptr) continue;
+
+      break;
+    }
+  }
+}
+
+void StatisticsWorker::_createCollection(std::string const& collection) const {
+  VPackBuilder s;
+  s.openObject();
+  s.add("name", VPackValue(collection));
+  s.add("isSystem", VPackValue(true));
+  s.add("waitForSync", VPackValue(false));
+
+  if (ServerState::instance()->isRunningInCluster() &&
+      ServerState::instance()->isCoordinator()) {
+    auto clusterFeature = application_features::ApplicationServer::getFeature<ClusterFeature>("Cluster");
+    s.add("replicationFactor", VPackValue(clusterFeature->systemReplicationFactor()));
+    s.add("distributeShardsLike", VPackValue("_graphs"));
+  }
+
+  s.add("journalSize", VPackValue(8 * 1024 * 1024));
+  s.close();
+
+  TRI_vocbase_t* vocbase = DatabaseFeature::DATABASE->systemDatabase();
+
+  int res = TRI_ERROR_NO_ERROR;
```

Comment:
This crashes on the coordinator.
Just start any test with `--cluster true`

We need to do a special stunt here with `createCollectionCoordinator` from `ClusterMethods` to make it run on the cluster.
Furthermore i would suggest to do this test vice versa:

Now:
Create collection => Test if it failed => Check for existing one on failure.

Better:
Check for existing collection => if not found => create new one and use it.

The suggested path is internally more performant